src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -106,62 +106,106 @@ static void shutdownClient(void)
     LM_E(("Database Shutdown Error %s (cannot shutdown mongo client)", status.toString().c_str()));
   }
 }
+
+
+
 /* ****************************************************************************
 *
-* mongoConnect -
+* mongoStart -
+*
+* This function must be called just once, because of the intialization of mongo::client
+* and the creation of the connection pool.
 */
-bool mongoConnect(const char*         host,
-                  const char*         db,
-                  const char*         rplSet,
-                  const char*         username,
-                  const char*         passwd,
-                  bool                _multitenant,
-                  double              timeout,
-                  int                 writeConcern)
+bool mongoStart
+(
+  const char* host,
+  const char* db,
+  const char* rplSet,
+  const char* username,
+  const char* passwd,
+  bool        _multitenant,
+  double      timeout,
+  int         writeConcern,
+  int         poolSize,
+  bool        semTimeStat
+)
 {
+  static bool called = false;
 
-    std::string err;
-    multitenant = _multitenant;
-
-    LM_T(LmtMongo, ("Connection info: dbName='%s', rplSet='%s', timeout=%f", db, rplSet, timeout));
+  if (called == true)
+  {
+    LM_W(("mongoStart already called - can only be called once, so please fix!"));
+    return false;
+  }
+  called = true;
 
-    mongoSemTake(__FUNCTION__, "connecting to mongo");
+  multitenant = _multitenant;
 
-    // We trust this function is called once. If not, this call
-    // should be protected against multiple calls
-    if (!clientIsInitialized)
+	if (!clientIsInitialized)
```

Comment:
`called` and `clientIsInitialized` seems to be for the same purpose. Why don't use just one?